lib/SILOptimizer/Transforms/SILMem2Reg.cpp

```diff
@@ -68,6 +68,19 @@ struct LiveValues {
   SILValue borrow = SILValue();
   SILValue copy = SILValue();
 
+  /// When iterating through the instructions in a block, whether, when the
+  /// storage stops storing a value, the lexical lifetime for that value should
+  /// be ended at that point in the execution of the pass.
+  ///
+  /// The lifetime should not be ended there if we don't yet have enough
+  /// information to end it.  That occurs when the running values are from a
+  /// load which was not preceeded by a store.  In that case, the lifetime end
+  /// will be added later, when we have enough information (the phi arguments)
+  /// to end it.
```

Comment:
It's hard to relate this discussion to the code. I think it's saying: return true if promoteAllocationInBlock has enough information to end the borrow scope. If this returns false, then the scope ending instructions will be emitting during fixBranchesAndUses

Reply:
How about

```
  /// Returns true if we have enough information to end the lifetime during
  /// promoteAllocationInBlock.
  ///
  /// The lifetime cannot be ended there if we don't yet have enough
  /// information to end it.  That occurs when the running values are from a
  /// load which was not preceeded by a store.  In that case, the lifetime end
  /// will be added later, when we have enough information (e.g., the phi
  /// arguments) to end it.
  bool canEndLexicalLifetimeInPromoteAllocationInBlock(AllocStackInst *asi) {
    return shouldAddLexicalLifetime(asi) && borrow;
  }
```